pkg/controllers/dashboard/fleetcharts/controller.go

```diff
@@ -57,9 +57,11 @@ func (h *handler) onSetting(key string, setting *v3.Setting) (*v3.Setting, error
 		return setting, nil
 	}
 
+	h.Lock()
 	if err := h.manager.Uninstall(fleetUninstallChart.ReleaseNamespace, fleetUninstallChart.ChartName); err != nil {
 		return nil, err
 	}
+	h.Unlock()
```

Comment:
would it better defer so we release the lock in case of error? or is it intentional to keep retrying for the same pod until it succeeds?

Reply:
I think it only needs to lock the uninstall place.... if using defer then we are locking the whole method and lock will only be released after the controller sync, which is not what we want?

Reply:
I see, adding Unlock() before returning error then? I think it'll be blocked otherwise

Reply:
:+1: